mod/node-core/pkg/builder/baseapp_options.go

```diff
@@ -98,3 +84,28 @@ func DefaultServiceOptions[
 		cometbft.SetChainID[LoggerT](chainID),
 	}
 }
+
+func loadChainIDFromGenesis(appOpts config.AppOptions) (string, error) {
+	var (
+		homeDir = cast.ToString(appOpts.Get(flags.FlagHome))
+		fp      = filepath.Join(homeDir, "config", "genesis.json")
+	)
+
+	f, err := os.Open(filepath.Clean(fp))
+	if err != nil {
+		return "", err
+	}
+
+	chainID, err := genutiltypes.ParseChainIDFromGenesis(f)
+	if err != nil {
+		return "",
+			errors.Join(
+				f.Close(),
+				fmt.Errorf(
+					"failed to parse chain-id from genesis file: %w",
+					err,
+				),
+			)
+	}
+	return chainID, f.Close()
```

Comment:
here we avoid defer f.Close() (which ignores the error causing the gosec warning) and we explicitly return the (unlikely) f.Close error to the caller.